WordPress/Classes/ViewRelated/Site Creation/Wizard/SiteCreator.swift

```diff
@@ -37,22 +37,49 @@ final class SiteCreator {
     /// - Returns: an Encodable object
     ///
     func build() throws -> SiteCreationRequest {
-        guard let domainSuggestion = address else {
+
+        guard let siteName = siteName else {
             throw SiteCreationRequestAssemblyError.invalidDomain
         }
-        let siteName = domainSuggestion.isWordPress ? domainSuggestion.subdomain : domainSuggestion.domainName
-        let siteDesign = design?.slug ?? "default"
+
+        let siteDesign = design?.slug ?? Strings.defaultDesignSlug
 
         let request = SiteCreationRequest(
             segmentIdentifier: segment?.identifier,
             siteDesign: siteDesign,
             verticalIdentifier: vertical?.slug,
-            title: information?.title ?? NSLocalizedString("Site Title", comment: "Site info. Title"),
+            title: information?.title ?? Strings.defaultSiteTitle,
             tagline: information?.tagLine ?? "",
             siteURLString: siteName,
-            isPublic: true
+            isPublic: true,
+            siteCreationFlow: address == nil ? Strings.siteCreationFlowForNoAddress : nil,
+            findAvailableUrl: address == nil
         )
 
         return request
     }
+
+    var hasSiteTitle: Bool {
+        information?.title != nil
+    }
+
+    /// Returns the domain suggestion if there's one,
+    /// - otherwise a site name if there's one,
+    /// - otherwise the account name if there's one,
+    /// - otherwise nil.
+    private var siteName: String? {
```

Comment:
How do you feel about renaming this to something like `siteURLString` instead? The site "name" versus site "title" has tripped me up multiple times now and I'm worried about the future. 😆

Reply:
We can rename it, but siteURLString would not be appropriate, since it does not necessarily contain a url. The parameter we pass to the API is `blog_name`. Would that sound better to you?

Reply:
So `blogName` (assuming we format it this way) == `siteURLString` (the `SiteCreationRequest` property) == `blog_name`? I just want to try to prevent confusion with "title" versus "name". I'm OK with whatever you decide.

Enums can be nice for mixed purpose types as well if we wanted to go that route.

Reply:
actually, since it maps to `siteURLString`, even if it might not contain a url, I'll just change to that